PublicMapping/district-builder-2
# Add service and endpoint to retrieve TopoJSON from S3

## Overview

Demonstrates how a NestJS singleton service can serve as a cache for data from S3.

The controller endpoint is ultimately likely to be unnecessary, but I added it to demonstrate how to use the TopologyService


### Demo

![image](https://user-images.githubusercontent.com/4432106/74751148-d8f90280-523a-11ea-8993-7f56d54793bf.png)


### Notes

This conflicts with changes in @rbreslow's PR. I'm planning to wait until his is merged to address those conflicts.

## Testing Instructions

- `scripts/setup`
- You should be able to download TopoJSON from http://localhost:3005/districts/topology/pa
- After the initial access, the above endpoint should continue to function even after disconnecting your computer's internet connection, because the value is cached.


Closes #25
